api/src/mutations/vault.rs

```diff
@@ -91,6 +148,86 @@ impl Mutation {
     }
 }
 
+/// Returns the ID of an existing wallet deduction entry if it exists for the given customer_treasury and asset_type.
+/// Otherwise, it generates a new pending deduction ID using the CreditsClient
+/// and creates a new entry in the wallet_deductions table, returning its ID.
+/// #Errors
+/// May return an error if there is an issue with querying or inserting data, or if the asset type is not supported.
+async fn submit_pending_deduction(
+    credits: &CreditsClient<Actions>,
+    db: &Connection,
+    balance: u64,
+    user_id: Uuid,
+    org_id: Uuid,
+    customer_treasury: Uuid,
+    asset_type: AssetType,
+) -> Result<Uuid> {
+    let wallet_deduction = WalletDeductions::find()
+        .filter(
+            wallet_deductions::Column::CustomerTreasury
+                .eq(customer_treasury)
+                .and(wallet_deductions::Column::AssetId.eq(asset_type)),
+        )
+        .one(db.get())
+        .await?;
+
+    if let Some(wallet_deduction) = wallet_deduction {
+        return Ok(wallet_deduction.id);
+    }
+
+    let id = match asset_type {
+        AssetType::Solana | AssetType::SolanaTest => {
+            credits
+                .submit_pending_deduction(
+                    org_id,
+                    user_id,
+                    Actions::CreateWallet,
+                    hub_core::credits::Blockchain::Solana,
+                    balance,
+                )
+                .await?
+        },
+        _ => {
+            return Err(Error::new("blockchain not supported yet"));
+        },
+    };
+
+    let deduction_id = id
+        .ok_or_else(|| Error::new("failed to generate credits deduction id"))?
+        .0;
+
+    let wallet_model = wallet_deductions::ActiveModel {
+        id: Set(deduction_id),
+        customer_treasury: Set(customer_treasury),
+        asset_id: Set(asset_type),
+        address: Set(None),
+        created_at: Set(Utc::now().into()),
+    };
```

Comment:
IDK not seeing lots of value on having a separate table for the deduction vs adding a column to the existing. 

I know it will null for projects but simpler mental modal.

Or if we have multiple tables why the extra fields does id, deduction_id not suffice on the table where id is the wallet id.